manifests/plugins.pp

```diff
@@ -0,0 +1,94 @@
+# @summary Manage Sensu plugins
+#
+# Class to manage the Sensu plugins.
+#
+# @example
+#   class { 'sensu::plugins':
+#     plugins    => ['disk-checks'],
+#     extensions => ['graphite'],
+#   }
+#
+# @example
+#   class { 'sensu::plugins':
+#     plugins    => {
+#       'disk-checks' => { 'version' => 'latest' },
+#     },
+#     extensions => {
+#       'graphite' => { 'version' => 'latest' },
+#     },
+#   }
+#
+# @param package_ensure
+#   Ensure property for sensu plugins package.
+# @param package_name
+#   Name of the Sensu plugins ruby package.
+# @param dependencies
+#   Package dependencies needed to install plugins and extensions.
+#   Default is OS dependent.
+# @param plugins
+#   Plugins to install
+# @param extensions
+#   Extensions to install
+#
+class sensu::plugins (
+  String $package_ensure = 'installed',
+  String $package_name = 'sensu-plugins-ruby',
+  Optional[Array] $dependencies = undef,
+  Optional[Variant[Array, Hash]] $plugins = undef,
+  Optional[Variant[Array, Hash]] $extensions = undef,
```

Comment:
Seems that VP is advocating using `[]` and `{}`. Should this be changed?

Reply:
I switched all 3 from using `Optional` to defaulting to empty Array.